### PR TITLE
[4302] Delete dialog submit fails with large submission comments

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/4302

- Delete dialog submit fails with large submission comments
- Updated delete api to v2

### CODE REVIEW

File: ui/app/src/modules/Content/Delete/DeleteDialog.tsx [hunk 1]
```diff
@@ -273,14 +314,20 @@ function DeleteDialogWrapper(props: DeleteDialogProps) {
     );
   }, [selectedItems, setApiState, siteId]);
 
+  useEffect(() => {
+    setSubmissionCommentCount({
+      currentLength: submissionComment.length
```
Comment (rart) [inline ui/app/src/modules/Content/Delete/DeleteDialog.tsx]: is there a reason to not use directly the comment to get it's length in place? that way we save some mem by dropping an additional state and effect.

File: ui/app/src/modules/Content/Delete/DeleteDialog.tsx [hunk 2]
```diff
@@ -41,11 +40,13 @@ import CircularProgress from '@material-ui/core/CircularProgress';
 import Dialog from '@material-ui/core/Dialog';
 import palette from '../../../styles/palette';
 import Grid from '@material-ui/core/Grid';
+import { Typography } from '@material-ui/core';
 
 interface DeleteDialogContentUIProps {
   resource: Resource<DeleteDependencies>;
   items: SandboxItem[];
   submissionComment: string;
+  submissionCommentCount: any;
```
Comment (rart) [inline ui/app/src/modules/Content/Delete/DeleteDialog.tsx]: type

File: ui/app/src/modules/Content/Delete/DeleteDialog.tsx [hunk 3]
```diff
@@ -55,6 +56,7 @@ interface DeleteDialogUIProps {
   items: SandboxItem[];
   selectedItems: SandboxItem[];
   submissionComment: string;
+  submissionCommentCount: any;
```
Comment (rart) [inline ui/app/src/modules/Content/Delete/DeleteDialog.tsx]: type

File: ui/app/src/modules/Content/Delete/DeleteDialog.tsx [hunk 4]
```diff
@@ -156,8 +169,31 @@ function DeleteDialogContentUI(props: DeleteDialogContentUIProps) {
               InputProps={{
                 className: classes.textField
               }}
+              inputProps={{ maxLength: submissionCommentCount.maxLength }}
```
Comment (rart) [inline ui/app/src/modules/Content/Delete/DeleteDialog.tsx]: submissionCommentCount is not a good name for the var that contains maxLength. also, consider just having a var for that specifically; what's the value of having it in an object with other stuff?

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/modules/Content/Delete/DeleteDialog.tsx [hunk 5]
```diff
@@ -235,11 +277,11 @@ export default function DeleteDialog(props: DeleteDialogProps) {
 function DeleteDialogWrapper(props: DeleteDialogProps) {
   const { items, onClose, onDismiss, onSuccess } = props;
   const [submissionComment, setSubmissionComment] = useState('');
+  const submissionCommentMaxLength = 250;
```
Comment (rart) [inline ui/app/src/modules/Content/Delete/DeleteDialog.tsx]: Let's move to GlobaState.env and supply to all dialogs with submission comments

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/modules/Content/Delete/DeleteDialog.tsx (hunk 5 above)
Comment (jvega190) [reply ui/app/src/modules/Content/Delete/DeleteDialog.tsx]: done, other dialogs will be done in here https://github.com/craftercms/craftercms/issues/4305

**Review by rart — APPROVED** [review]

**Pull request merged**